generator/ServiceClientGeneratorLib/Generators/Marshallers/JsonRPCStructureMarshaller.tt

```diff
@@ -207,25 +200,126 @@ namespace <#=        this.Config.Namespace #>.Model.Internal.MarshallTransformat
 				
 			}
 	}
+
+	protected void ProcessMap(int level, string variableName, Shape structure)
+	{
+		string flatVariableName = variableName.Replace(".", "");
+#>
+<#=new string(' ', level * 4)#>            context.Writer.WriteStartObject();
+<#=new string(' ', level * 4)#>            foreach (var <#=flatVariableName#>Kvp in <#=variableName#>)
+<#=new string(' ', level * 4)#>            {
+<#=new string(' ', level * 4)#>                context.Writer.WritePropertyName(<#=flatVariableName#>Kvp.Key);
+<#=new string(' ', level * 4)#>                var <#=flatVariableName#>Value = <#=flatVariableName#>Kvp.Value;
+
+<#+             // Check for null values - only null checks for sparse maps as defined in customizations
+
+                bool isNullableMap = false;
+                if (this.Structure != null)
+                {
+                    var memberWithMap = this.Structure.Members.FirstOrDefault(m => m.PropertyName == variableName.Split('.').Last());
+                    isNullableMap = (memberWithMap != null && memberWithMap.UseNullable);
```

Comment:
Not blocking but another slightly more compact way to do this is: `isNullableMap = memberWithMap?.UseNullable == true;`